### PR TITLE
new responsive site

### LINKED ISSUE #131: new responsive site
i made a copy of the current site and ported it to a full responsive site:
[https://edswf1.github.io/dogecoin.com/](https://edswf1.github.io/dogecoin.com/)

i will add modals and tutorial tomorrow.
other todos are:
- resize images
- new mockup for dogecoin wallet
- seo
- and maybe some new text content?

### PR DESCRIPTION
**Pull request by edswf1**
this site is a fully functional, responsive version of the current dogecoin.com site base on Bootstrap and jQuery.
live version of the site: https://edswf1.github.io/dogecoin.com/
i18n is not included yet, however all other functionality, including tutorial, is.

All images, js and css resources where moved to an assets folder, dogechain api is used to display pricing.

This site has been tested for perfomance, see mobile score below.
perfomance

Libraries used:
- Bootstrap 4.5.3
- jQuery 3.5.1
- jQuery Easing 1.4.1
- FontAwesome 5.15.0
- particles.js
- Google Fonts Montserrat & Lato

i did not remove any project files, only index.html would be replaced

### CODE REVIEW

**Comment by patricklodder** [comment]
Hi! Thanks! How does this integrate with the internationalization?

**Comment by edswf1** [comment]
> Hi! Thanks! How does this integrate with the internationalization?

unfortunately it doesn‘t right now. i‘m planning to integrate i18n tomorrow

**Comment by edswf1** [comment]
@patricklodder 
added internationalization with jQuery.i18n.
Language is guessed by browser language (english fallback) and can be set in footer.
Once Language is set, current Language ist stored in sessionStorage (for maintaining language when browsing to guide.html)
added some texts to en,de,fr and es.json, however im not fluent in fr and es.

site is live at: https://edswf1.github.io/dogecoin.com-1/

**Comment by patricklodder** [comment]
Thank you! To keep momentum going, I'm going to try to give you feedback in iterations. This is going to make me look like a terrible nitpicker that is never satisfied 😂 - but the review is too big a task to go all-at-once.

Remarks for now:
- Could you please delete unused old files, so that git's diff recognizes the file move and any subsequent changes, i.e.: `./imgs` and `./i18n`
- Could you please separate the ticker from this pull request and if you really think this should be done, PR it separately, later? This is a controversial change and we should not want discussion on that block this.
- What is the reason for `css/styles.css` to include bootstrap? I normally advice against that because the beauty of the c (cascading) part in css is that we don't need to couple - so the default of having stylesheets cascade means ease of maintenance.
- Related to the point above: Is it possible to split and mark untouched "vendor" assets from bespoke assets?
- is there a reason why some things (`./imgs`, `./i18n`) moved to `./assets` while others (like `./js` and `css`) did not?

**Comment by edswf1** [comment]
> Thank you! To keep momentum going, I'm going to try to give you feedback in iterations. This is going to make me look like a terrible nitpicker that is never satisfied 😂 - but the review is too big a task to go all-at-once.
> 
> Remarks for now:
> 
> * Could you please delete unused old files, so that git's diff recognizes the file move and any subsequent changes, i.e.: `./imgs` and `./i18n`
> * Could you please separate the ticker from this pull request and if you really think this should be done, PR it separately, later? This is a controversial change and we should not want discussion on that block this.
> * What is the reason for `css/styles.css` to include bootstrap? I normally advice against that because the beauty of the c (cascading) part in css is that we don't need to couple - so the default of having stylesheets cascade means ease of maintenance.
> * Related to the point above: Is it possible to split and mark untouched "vendor" assets from bespoke assets?
> * is there a reason why some things (`./imgs`, `./i18n`) moved to `./assets` while others (like `./js` and `css`) did not?

No Problem:
- deleted unused old files but kept getting started folder for .md files
- removed the ticker. 1 Dogecoin = 1 Dogecoin.
- seperated bootstrap.css and styles.css
- moved all vendor assets to assets/vendor/ . all contents of this folder are untouched
- there was no reason for keeping js and css outside of assets folder. fixed that

File: assets/i18n/en.json [hunk 1]
```diff
@@ -247,5 +247,15 @@
...
+    "dgc-configure-wallet": "Configure your Wallet",
+    "dgc-configure-wallet-1": "After downloading, you can configure your Wallet according to our <a class=\"btn btn-sm btn-outline-secondary\" href=\"guide.html\">Guide</a>",
+    "dgc-choose-wallet-1": "Pick a Wallet from our <a class=\"btn btn-sm btn-outline-secondary js-scroll-trigger\" href=\"#wallets\">Wallet Section</a>",
+    "dgc-choose-wallet-2": "A Wallet is used for holding your Dogecoins directly on your Computer/Smartphone.",
+    "dgc-guide": "Guide",
+    "dgc-menu": "Menu",
+    "dgc-get-some": "Get Some Dogecoin",
+    "dgc-language": "Language",
+    "dgc-back-to-main": "Back to Main Page"
+}
```
Comment (patricklodder) [inline assets/i18n/en.json]: please restore the newline

File: assets/css/styles.css [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  .pr-lg-3,
+  .px-lg-3 {
+    padding-right: .5rem !important;
+  }
+
+  .pl-lg-3,
+  .px-lg-3 {
+    padding-left: .5rem !important;
+  }
+}
```
Comment (patricklodder) [inline assets/css/styles.css]: please add a newline

File: assets/js/scripts.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+     var commas = 6;
+     if(currency == 'BTC'){
+      commas = 8;
+    }
+    var price = parseFloat(response.data.prices[0].price);
+    var priceBase = response.data.prices[0].price_base;
+    $selector = $(".curPrice_"+currency);
+    $selector.html(price.toFixed(commas) + " " + priceBase);
+  });
+  }
```
Comment (patricklodder) [inline assets/js/scripts.js]: this should be removed

File: guide.html [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+        <div class="container-fluid">
+            <a class="navbar-brand js-scroll-trigger" href="#page-top"><img src="assets/img/dogecoin-300.png">&nbsp;Dogecoin</a>
+            <button class="navbar-toggler navbar-toggler-right text-uppercase font-weight-bold bg-primary text-secondary rounded" type="button" data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
+                Menu
+                <i class="fas fa-bars"></i>
+            </button>
+            <div class="collapse navbar-collapse" id="navbarResponsive">
+                <ul class="navbar-nav ml-auto">
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded" href="index.html" data-i18n="dgc-back-to-main">Back to Main Page</a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><span class="nav-link py-3 px-0 px-lg-3 rounded text-white curPrice_USD"></span></li>
```
Comment (patricklodder) [inline guide.html]: this should be removed

File: guide.html [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+    <script src="assets/vendor/js/jquery.i18n/languages/he.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/hsb.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/hu.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/hy.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/la.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/ml.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/os.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/ru.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/sl.js"></script>
+    <script src="assets/vendor/js/jquery.i18n/languages/uk.js"></script>
```
Comment (patricklodder) [inline guide.html]: do all of these need to be loaded even if we don't have translations? If yes, can we minify this into a single script?

File: index.html [hunk 6]
```diff
@@ -1,556 +1,290 @@
...
+                <span data-i18n="dgc-menu">Menu</span>
+                <i class="fas fa-bars"></i>
+            </button>
+            <div class="collapse navbar-collapse" id="navbarResponsive">
+                <ul class="navbar-nav ml-auto">
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#wallets">Wallets</a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#about"><span data-i18n="dgc-what-is-doge">What is dogecoin</span></a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#start"><span data-i18n="dgc-get-started-now">How to get started</span></a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded" href="guide.html"><span data-i18n="dgc-guide">Guide</span></a></li>
+                    <li class="nav-item mx-0 mx-lg-1"><a href="" class="nav-link py-3 px-0 px-lg-3 rounded text-white curPrice_USD"></a></li>
```
Comment (patricklodder) [inline index.html]: this needs to be removed

**Review by patricklodder — COMMENTED** [review]
went through all the files, have some minor comments.

File: guide.html (hunk 5 above)
Comment (edswf1) [reply guide.html]: no these files where never used (imho only needed for special characters), just included them for the sake of completeness (as they are included in the current live page). removed them now form index.html and guide.html

File: index.html (hunk 6 above)
Comment (edswf1) [reply index.html]: i think you are referring to the curPrice_USD element. removed that now. all other elements are part of the menu and shouldn't be removed

File: guide.html (hunk 4 above)
Comment (edswf1) [reply guide.html]: i think you are referring to the curPrice_USD element. removed that now. "Back to Main Page" is part of the menu and shouldn't be removed

File: assets/js/scripts.js (hunk 3 above)
Comment (edswf1) [reply assets/js/scripts.js]: forgot that. removed it now

File: assets/i18n/en.json (hunk 1 above)
Comment (edswf1) [reply assets/i18n/en.json]: unfortunately i don't know where to restore a newline here.
Edit: sry, got it. added newline on end of file.

File: assets/css/styles.css (hunk 2 above)
Comment (edswf1) [reply assets/css/styles.css]: unfortunately i don't know where to add newline here.
Edit: sry got it! added new line on end of file

**Comment by edswf1** [comment]
> went through all the files, have some minor comments.

hope i resolved the issues, menu items are now free of pricing.
sorry for the newline stuff, never heard of that convention before.

**Comment by lugus** [comment]
> @patricklodder
> added internationalization with jQuery.i18n.
> Language is guessed by browser language (english fallback) and can be set in footer.
> Once Language is set, current Language ist stored in sessionStorage (for maintaining language when browsing to guide.html)
> added some texts to en,de,fr and es.json, however im not fluent in fr and es.
> 
> site is live at: https://edswf1.github.io/dogecoin.com-1/

French LGTM! 💯 Thanks for your work @edswf1 @patricklodder 🚀

**Review by michilumin — COMMENTED** [review]
So I've been looking this over and, first of all, it's great, I really want to see it up soon!  The following are just ideas/suggestions though;  doing it in review just to have eyes on things:

-  /r/dogecoin isn't the only community center: "community" should link not only to reddit /r/dogecoin, but also to discord ( https://discord.gg/dogecoin ) and /r/dogeducation.
-  The only stylistic wonderings I have is the 'star' motif, since especially when it's blue, it looks very ... "american" - to a lot of people at least.  This isn't a big thing, I'm just wondering if we could replace the star with something else.  (moon?)
-  Pedantry time!  OSX isn't OSX anymore, it's now officially MacOS, so we should probably update wording to that.

... Finally, I know for a while we had a source for a dev ticker of sorts. In fact I think Langer or Ross had it working, showing the last few commits from github. I really do want to revisit this but I also really really DONT want to delay this thing going up. Because it is VERY VERY good and I don't want to get in the way - I just want to touch on that point and see how possible such a 'github repo ticker' as an element could be.

**Comment by edswf1** [comment]
> So I've been looking this over and, first of all, it's great, I really want to see it up soon! The following are just ideas/suggestions though; doing it in review just to have eyes on things:
> 
> * /r/dogecoin isn't the only community center: "community" should link not only to reddit /r/dogecoin, but also to discord ( https://discord.gg/dogecoin ) and /r/dogeducation.
> * The only stylistic wonderings I have is the 'star' motif, since especially when it's blue, it looks very ... "american" - to a lot of people at least.  This isn't a big thing, I'm just wondering if we could replace the star with something else.  (moon?)
> * Pedantry time!  OSX isn't OSX anymore, it's now officially MacOS, so we should probably update wording to that.
> 
> ... Finally, I know for a while we had a source for a dev ticker of sorts. In fact I think Langer or Ross had it working, showing the last few commits from github. I really do want to revisit this but I also really really DONT want to delay this thing going up. Because it is VERY VERY good and I don't want to get in the way - I just want to touch on that point and see how possible such a 'github repo ticker' as an element could be.

Thank you for your feedback. 
- added discord community at about section and footer
- changed star to angle-down (basically any font-awesome icon or image could be there)
- changed OS X to macOS

regarding github repo ticker: implementing this would be an easy task (simply parsing https://api.github.com/repos/dogecoin/dogecoin/commits response), however i would prefer doing this in a succeeding PR

**Comment by patricklodder** [comment]
> regarding github repo ticker: implementing this would be an easy task (simply parsing https://api.github.com/repos/dogecoin/dogecoin/commits response), however i would prefer doing this in a succeeding PR

This is my preferred strategy too.

---

Two last comments before I do a final round of code review:

1. The Dogecoin coin logo changed into a Doge image, without the coin... why?
2. The creator credits at the bottom are gone. Intentional?

**Comment by edswf1** [comment]
> > regarding github repo ticker: implementing this would be an easy task (simply parsing https://api.github.com/repos/dogecoin/dogecoin/commits response), however i would prefer doing this in a succeeding PR
> 
> This is my preferred strategy too.
> 
> Two last comments before I do a final round of code review:
> 
> 1. The Dogecoin coin logo changed into a Doge image, without the coin... why?
> 2. The creator credits at the bottom are gone. Intentional?

1. i basically took the image of the current "what is doge" section as hero image because the coin itself is flying around in the background, is part of the menu and the favicon. Would be no problem to change that though.
2. added footer-credits again. please let me know if there should be other mentions too

**Comment by patricklodder** [comment]
> 1. i basically took the image of the current "what is doge" section as hero image because the coin itself is flying around in the background, is part of the menu and the favicon. Would be no problem to change that though.

No this is fine, just wanted to know why.

> 2. added footer-credits again. please let me know if there should be other mentions too

Thanks! No further mentions required I think. I'll go over the 6 files that have been changed once more today.

**Comment by lugus** [comment]
Looks like http://www.ummjackson.com is not valid anymore and should be replaced by https://ummjackson.com/

**Comment by edswf1** [comment]
> Looks like http://www.ummjackson.com is not valid anymore and should be replaced by https://ummjackson.com/

thanks, fixed that

**Comment by nickatnight** [comment]
Suggestion:

Should there be a link somewhere on the page that points back to this repo?

**Comment by patricklodder** [comment]
> Should there be a link somewhere on the page that points back to this repo?

I thought about that and I will propose a separate PR with that change after we merge this.

**Review by patricklodder — DISMISSED** [review]
Full ACK. Reviewed all content & scripts. Checked all links. All things I don't like are inherited from the current site, so we will handle those outside of this PR, as they have nothing to do with the updated design.

I'm leaving this open for 12 hours before deployment (3/30, 8am PDT)

**Comment by nickatnight** [comment]
@patricklodder just in time for the moon boost 💎 🤲

**Comment by nickatnight** [comment]
Swell job @edswf1

File: assets/i18n/en.json [hunk 7]
```diff
@@ -53,7 +53,7 @@
     "dgc-get-dogecoin-how": "Find out how to get Dogecoin.",
     "dgc-stores-title": "Stores.", 
     "dgc-stores-buy": "Buy with Dogecoin.",
-    "dgc-footer-credits": "Created by <a href=\"http://www.ummjackson.com\" target=\"_blank\">Jackson Palmer</a> & <a href=\"https://twitter.com/BillyM2k\" target=\"_blank\">Shibetoshi Nakamoto</a>. Website Design by <a href=\"https://www.reddit.com/user/PowerLemons\" target=\"_blank\">/u/PowerLemons</a>. Logo by <a href=\"http://christinemix.com\" target=\"_blank\">Christine Ricks</a>.",
+    "dgc-footer-credits": "Created by <a href=\"http://www.ummjackson.com\" target=\"_blank\">Jackson Palmer</a> & <a href=\"https://twitter.com/BillyM2k\" target=\"_blank\">Shibetoshi Nakamoto</a>. Website Design by <a href=\"https://www.reddit.com/user/edswf1\" target=\"_blank\">/u/edswf1</a>. Logo by <a href=\"http://christinemix.com\" target=\"_blank\">Christine Ricks</a>.",
```
Comment (lugus) [inline assets/i18n/en.json]: old url `http://www.ummjackson.com` here and in other translation files

File: assets/js/scripts.js [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+  i18n.load('assets/i18n', i18n.locale).done(
+    function() {
+     $('[data-i18n]').each(function(index) {
+      $(this).i18n();
+      $(this).html($.i18n($(this).attr('data-i18n')));
+    } );
+   } );
+}
+// Enable debug
+$.i18n.debug = true;
```
Comment (lugus) [inline assets/js/scripts.js]: I think we should disable the debug mode?

File: assets/i18n/en.json (hunk 7 above)
Comment (patricklodder) [reply assets/i18n/en.json]: I saw those, those are also on the current site. There's also a couple more http links in translation files that need to be switched to https. I'll prepare a PR before launch so we can keep this going.

File: assets/js/scripts.js (hunk 8 above)
Comment (patricklodder) [reply assets/js/scripts.js]: good catch... I missed that one, thanks! I'll PR that separately.

Comment (edswf1) [reply assets/js/scripts.js]: > good catch... I missed that one, thanks! I'll PR that separately.

sorry missed your comment and updated already

File: assets/i18n/en.json (hunk 7 above)
Comment (edswf1) [reply assets/i18n/en.json]: > I saw those, those are also on the current site. There's also a couple more http links in translation files that need to be switched to https. I'll prepare a PR before launch so we can keep this going.

sorry missed your comment and updated already

**Review by patricklodder — APPROVED** [review]
Checked delta for commits ea6fb289fca323e3089a2e09be94ea26991c84b3 and 39318a4859e45321574c1204475ce1f3551fde61 since my last review. OK.

**Comment by lugus** [comment]
Kudos to @edswf1 👏

**Pull request merged**